src/ferny/transport.py

```diff
@@ -0,0 +1,371 @@
+# ferny - asyncio SSH client library, using ssh(1)
+#
+# Copyright (C) 2023 Allison Karlitskaya <allison.karlitskaya@redhat.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import typing
+from typing import Any, Callable, Iterable, Sequence, TypeVar
+
+from .interaction_agent import InteractionAgent, InteractionError, InteractionHandler, get_running_loop
+from .ssh_errors import get_exception_for_ssh_stderr
+
+logger = logging.getLogger(__name__)
+
+P = TypeVar('P', bound=asyncio.Protocol)
+
+
+class FernyTransport(asyncio.Transport, asyncio.SubprocessProtocol):
+    # These get initialized in connection_made(), not __init__(), but we don't
+    # list them as Optional because we won't return the object to the user
+    # until after the initialization is complete.
+    _subprocess_transport: 'asyncio.SubprocessTransport | None' = None
+    _stdin_transport: 'asyncio.WriteTransport | None' = None
+    _stdout_transport: 'asyncio.ReadTransport | None' = None
```

Comment:
The comment is confusing. "We don't list them as Optional", and then the code does exactly that (well, `x | None` == `Optional[x]`)

Reply:
This is a leftover from a simpler time.  Thanks for spotting it.